tests/core/test_core.py

```diff
@@ -427,6 +427,8 @@ def test_open_in_browser_success(
     ) -> None:
         # Set DISPLAY environ to be able to run test in CI
         os.environ["DISPLAY"] = ":0"
+        # Set TERM environ to be able to check text-browsers
+        os.environ["TERM"] = "xterm-256color"
```

Comment:
Your test changes make the tests pass, but don't test the expectations from the new behavior:
- is the environment variable unchanged?
- can we test specifically what happens if there is no 'valid' webbrowser found?

Reply:
I'm a bit confused about writing the tests for this. The `TERM` environ doesn't really change, how do we test its transitory absence?
Also writing a test specifically for text_browser sounds good but again not sure what we can test. We can assert if `TERM` environ doesn't exist inos.environ but that's a value we are setting and not what we are fetching from the function.